### PR TITLE
feat: Add uncompletable task support with isUncompletable field

### PR DESCRIPTION
**Pull request by scottlovegrove**
## Summary
Adds support for uncompletable tasks through a new `isUncompletable` boolean field. This feature allows programmatic creation of tasks that cannot be completed, which appear as organizational headers in Todoist.

## What Changed
- **New Field**: Added optional `isUncompletable?: boolean` to `AddTaskArgs`, `UpdateTaskArgs`, and `QuickAddTaskArgs`
- **Auto-Detection**: TaskSchema now auto-detects tasks with `* ` prefix and sets `isUncompletable: true`
- **Content Processing**: Added utilities in `uncompletable-helpers.ts` to handle `* ` prefix logic
- **API Integration**: Updated `addTask()`, `updateTask()`, and `quickAddTask()` methods to process content based on the flag
- **Comprehensive Testing**: 36 unit tests + 7 integration tests covering all edge cases

## Behavior
- When `isUncompletable: true` → adds `* ` prefix to content (if not already present)
- Content with existing `* ` prefix takes precedence over field value (backward compatibility)
- Auto-detection ensures existing uncompletable tasks work seamlessly

## Usage Examples
```typescript
// Create uncompletable task
const task = await api.addTask({
    content: 'Project planning session',
    isUncompletable: true
});
// Result: content = "* Project planning session", isUncompletable = true

// Update task to be uncompletable
await api.updateTask('123', {
    content: 'Updated header',
    isUncompletable: true
});

// Works with quickAddTask too
await api.quickAddTask({
    text: 'Quick organizational header',
    isUncompletable: true
});
```

## Testing
- ✅ All 273 existing tests pass (no regressions)
- ✅ 43 new tests added (36 unit + 7 integration)
- ✅ Covers content processing, API integration, edge cases, and backward compatibility

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/utils/uncompletable-helpers.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    }
+    return UNCOMPLETABLE_PREFIX + content
+}
+
+/**
+ * Removes the uncompletable prefix (* ) from task content if present
+ * @param content - The task content
+ * @returns Content with uncompletable prefix removed
+ */
+export function removeUncompletablePrefix(content: string): string {
```
[the remark this answers is not in the document]
Comment (scottlovegrove) [reply src/utils/uncompletable-helpers.ts]: ~Thanks, will remove.~

Actually, I'll leave it as it's a helper function that could be useful for external use.

File: src/utils/uncompletable-helpers.test.ts [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+        test('returns false for content without prefix', () => {
+            expect(hasUncompletablePrefix('Regular task')).toBe(false)
+        })
+
+        test('returns false for asterisk without space', () => {
+            expect(hasUncompletablePrefix('*No space')).toBe(false)
+        })
+
+        test('returns true for just the prefix', () => {
+            expect(hasUncompletablePrefix('* ')).toBe(true)
```
[the remark this answers is not in the document]
Comment (pedroalves0) [reply src/utils/uncompletable-helpers.test.ts]: The reason is that if `* ` is uncompletable, then the content becomes `''` as the uncompletable prefix is `'* '`, which isn’t valid task content.
